plugins/destination/postgresql/client/read.go

```diff
@@ -113,24 +141,31 @@ func (c *Client) reverseTransform(f arrow.Field, bldr array.Builder, val any) er
 	return nil
 }
 
-func (c *Client) reverseTransformer(sc *arrow.Schema, values []any) (arrow.Record, error) {
+func (c *Client) reverseTransformer(table *schema.Table, values []any) (arrow.Record, error) {
+	sc := table.ToArrowSchema()
 	bldr := array.NewRecordBuilder(memory.DefaultAllocator, sc)
 	for i, f := range sc.Fields() {
-		if err := c.reverseTransform(f, bldr.Field(i), values[i]); err != nil {
-			return nil, err
+		if c.pgType == pgTypePostgreSQL {
+			if err := c.reverseTransform(f, bldr.Field(i), values[i]); err != nil {
+				return nil, err
+			}
+		} else {
+			if err := c.reverseTransformCockroach(f, bldr.Field(i), values[i]); err != nil {
+				return nil, err
+			}
 		}
 	}
 	rec := bldr.NewRecord()
 	return rec, nil
 }
 
-func (c *Client) Read(ctx context.Context, table *arrow.Schema, sourceName string, res chan<- arrow.Record) error {
-	colNames := make([]string, 0, len(table.Fields()))
-	for _, col := range table.Fields() {
+func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName string, res chan<- arrow.Record) error {
+	colNames := make([]string, 0, len(table.Columns))
+	for _, col := range table.Columns {
 		colNames = append(colNames, pgx.Identifier{col.Name}.Sanitize())
 	}
```

Comment:
```suggestion
	colNames := make([]string, len(table.Columns))
	for i, col := range table.Columns {
		colNames[i] = pgx.Identifier{col.Name}.Sanitize()
	}
```

Reply:
I think it's identitcal

Reply:
Unfortunately, append isn't identical to plain assignment by index, as it branches to verify the current length.
It can be somewhat summarized to the following piece of code:
```go
func append[A any](dst []A, elems...A) []A {
	l, c := len(dst), cap(dst)

	if l + len(elems) > c {
		// realloc dst, with capacity = 2*c
	}

	res = dst[0:l+len(elems)]
	for i := 0; i < len(elems); i++ {
		res[l+i] = elems[i]
	}
	return res
}
```

So, when you have an option to use the assignment by index, it's better to stick to it performance-wise.

Reply:
However, in this case it's a nit